codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/middleware/MiddlewareStackStep.java

```diff
@@ -42,8 +42,8 @@ public String toString() {
                 return "Retry";
             case SERIALIZE:
                 return "Serialize";
-            case BUILD:
-                return "Build";
+            case VALIDATE:
```

Comment:
Build is confusing with serialize. I think we needed one at the validate step.

Reply:
These steps were intended to match those used by other SDKs. See for example
* https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/modules/_aws_sdk_middleware_stack.html
* https://aws.github.io/aws-sdk-go-v2/docs/middleware/#:~:text=You%20can%20customize%20AWS%20SDK,step's%20input%20and%20output%20types.

The meanings were intended to be:
* Serialize - Serialize the given parameter input into a request data structure that can be confusmed by the transport layer (eg, an HTTP:Request).
* Build - Provide additional transformation of the Request after serialization (eg ContentLength)

I agree however that because of our choice to use "Builders" for Serialization that this is a bit confusing.  I don't think VALIDATE is the right name for this step though.

Reply:
I just undid it. Can re-evaluate later.